File: Flutter-conference-speakup/lib/core/apis/endpoints.dart
/// Centralized API endpoint paths matching backend routes.
/// All paths are relative — base URL is set on the Dio client.
class ApiEndpoints {
  ApiEndpoints._();

  // ──────────── Auth ────────────
  static const signIn = '/auth/signin';
  static const signOut = '/auth/signout';
  static const me = '/auth/me';
  static const deleteAccount = '/auth/account';

  // ──────────── Users ────────────
  static const userProfile = '/users/profile';
  static const userAvatar = '/users/avatar';
  static const userDevices = '/users/devices';
  static String userDevice(String deviceId) => '/users/devices/$deviceId';
  static const userOnlineStatus = '/users/online-status';

  // ──────────── Meetings ────────────
  static const meetings = '/meetings';
  static String meeting(String id) => '/meetings/$id';
  static String joinByCode(String code) => '/meetings/join/$code';
  static String joinMeeting(String id) => '/meetings/$id/join';
  static String leaveMeeting(String id) => '/meetings/$id/leave';
  static String endMeeting(String id) => '/meetings/$id/end';
  static String lockMeeting(String id) => '/meetings/$id/lock';
  static String unlockMeeting(String id) => '/meetings/$id/unlock';
  static String meetingParticipants(String id) => '/meetings/$id/participants';
  static String kickParticipant(String meetingId, String participantId) =>
      '/meetings/$meetingId/kick/$participantId';
  static String meetingToken(String id) => '/meetings/$id/token';

  // ──────────── Rooms ────────────
  static const activeRooms = '/rooms/active';
  static String roomState(String id) => '/rooms/$id';
  static String roomSettings(String id) => '/rooms/$id/settings';
  static String roomMuteAll(String id) => '/rooms/$id/mute-all';

  // ──────────── Chat ────────────
  static const chatRooms = '/chat/rooms';
  static String meetingChat(String meetingId) => '/chat/meeting/$meetingId';
  static String chatMessages(String chatRoomId) => '/chat/$chatRoomId/messages';
  static String deleteMessage(String messageId) => '/chat/messages/$messageId';

  // ──────────── Notifications ────────────
  static const notifications = '/notifications';
  static const notificationUnreadCount = '/notifications/unread-count';
  static const notificationReadAll = '/notifications/read-all';
  static String notificationRead(String id) => '/notifications/$id/read';
  static String notificationDelete(String id) => '/notifications/$id';

  // ──────────── Recordings ────────────
  static const recordings = '/recordings';
  static String recording(String id) => '/recordings/$id';
  static String recordingDownload(String id) => '/recordings/$id/download';
  static String startRecording(String meetingId) =>
      '/recordings/meeting/$meetingId/start';
  static String stopRecording(String meetingId) =>
      '/recordings/meeting/$meetingId/stop';

  // ──────────── Analytics ────────────
  static const analyticsDashboard = '/analytics/dashboard';
  static const analyticsUsage = '/analytics/usage';
  static String meetingAnalytics(String meetingId) =>
      '/analytics/meeting/$meetingId';

  // ──────────── Billing ────────────
  static const subscription = '/billing/subscription';
  static const billingCheckout = '/billing/checkout';
  static const billingPortal = '/billing/portal';
  static const billingCancel = '/billing/cancel';

  // ──────────── Search ────────────
  static const search = '/search';
  static const searchUsers = '/search/users';
  static const searchMeetings = '/search/meetings';
}

File: webapp-conference-speakup/src/config/endpoints.ts
/** API endpoint paths — mirrors Flutter ApiEndpoints exactly */
export const endpoints = {
  // ─── Auth ───
  auth: {
    signIn: "/auth/signin",
    signOut: "/auth/signout",
    me: "/auth/me",
    deleteAccount: "/auth/account",
  },

  // ─── Users ───
  users: {
    profile: "/users/profile",
    avatar: "/users/avatar",
    devices: "/users/devices",
    device: (id: string) => `/users/devices/${id}`,
    onlineStatus: "/users/online-status",
  },

  // ─── Meetings ───
  meetings: {
    list: "/meetings",
    byId: (id: string) => `/meetings/${id}`,
    joinByCode: (code: string) => `/meetings/join/${code}`,
    join: (id: string) => `/meetings/${id}/join`,
    leave: (id: string) => `/meetings/${id}/leave`,
    end: (id: string) => `/meetings/${id}/end`,
    lock: (id: string) => `/meetings/${id}/lock`,
    unlock: (id: string) => `/meetings/${id}/unlock`,
    participants: (id: string) => `/meetings/${id}/participants`,
    kick: (meetingId: string, participantId: string) =>
      `/meetings/${meetingId}/kick/${participantId}`,
    token: (id: string) => `/meetings/${id}/token`,
  },

  // ─── Rooms ───
  rooms: {
    active: "/rooms/active",
    state: (id: string) => `/rooms/${id}`,
    settings: (id: string) => `/rooms/${id}/settings`,
    muteAll: (id: string) => `/rooms/${id}/mute-all`,
  },

  // ─── Chat ───
  chat: {
    rooms: "/chat/rooms",
    meetingChat: (meetingId: string) => `/chat/meeting/${meetingId}`,
    messages: (chatRoomId: string) => `/chat/${chatRoomId}/messages`,
    deleteMessage: (messageId: string) => `/chat/messages/${messageId}`,
  },

  // ─── Notifications ───
  notifications: {
    list: "/notifications",
    unreadCount: "/notifications/unread-count",
    readAll: "/notifications/read-all",
    read: (id: string) => `/notifications/${id}/read`,
    delete: (id: string) => `/notifications/${id}`,
  },

  // ─── Recordings ───
  recordings: {
    list: "/recordings",
    byId: (id: string) => `/recordings/${id}`,
    download: (id: string) => `/recordings/${id}/download`,
    start: (meetingId: string) => `/recordings/meeting/${meetingId}/start`,
    stop: (meetingId: string) => `/recordings/meeting/${meetingId}/stop`,
  },

  // ─── Analytics ───
  analytics: {
    dashboard: "/analytics/dashboard",
    usage: "/analytics/usage",
    meeting: (meetingId: string) => `/analytics/meeting/${meetingId}`,
  },

  // ─── Billing ───
  billing: {
    subscription: "/billing/subscription",
    checkout: "/billing/checkout",
    portal: "/billing/portal",
    cancel: "/billing/cancel",
  },

  // ─── Search ───
  search: {
    global: "/search",
    users: "/search/users",
    meetings: "/search/meetings",
  },
} as const;
